cmd/archeio/app/handlers.go

```diff
@@ -97,47 +98,49 @@ func makeV2Handler(rc RegistryConfig, blobs blobChecker) func(w http.ResponseWri
 			return
 		}
 
-		// check if blob request
-		matches := reBlob.FindStringSubmatch(path)
-		if len(matches) != 2 {
-			// not a blob request so forward it to the main upstream registry
-			redirectPath := calculateRedirectPath(rc, path)
-			klog.V(2).InfoS("redirecting manifest request to upstream registry", "path", path, "redirect", rc.UpstreamRegistryEndpoint+redirectPath)
-			http.Redirect(w, r, rc.UpstreamRegistryEndpoint+redirectPath, http.StatusTemporaryRedirect)
-			return
-		}
-
-		// for blob requests, check the client IP and determine the best backend
-		clientIP, err := getClientIP(r)
-		if err != nil {
-			// this should not happen
-			klog.ErrorS(err, "failed to get client IP")
-			http.Error(w, err.Error(), http.StatusBadRequest)
-			return
-		}
+		if rc.ServeImagesfromAWS {
```

Comment:
TODO(minor): this is only set at startup and we're returning a request handler function, we can avoid the runtime check on each request.